app/models/conversion_host.rb

```diff
@@ -124,28 +208,61 @@ def resource_supports_conversion_host
     end
   end
 
-  def check_resource_credentials(fatal = false, extra_msg = nil)
-    success = send("check_resource_credentials_#{resource.ext_management_system.emstype}")
-    if !success and fatal
-      msg = "Credential not found for #{resource.name}."
-      msg += " #{extra_msg}" unless extra_msg.blank?
-      _log.error(:msg)
+  # Find the associated credentials for the current conversion host based
+  # on the type of provider. Each provider should look for a directly
+  # associated authentication first, and default to the authentication
+  # for the resource second.
+  #--
+  # TODO: Each provider subclass should define their own method for
+  # finding credentials.
+  #
+  def find_credentials(msg = nil)
+    authentication = send("find_resource_credentials_#{resource.ext_management_system.emstype}")
```

Comment:
Can we just do this in a provider subclass now?

Reply:
That's the plan eventually, want to leave it as a TODO for now.

Reply:
Actually, this was simplified so that it's no longer provider dependent. Though, of course, it could be redefined if ever needed.

Reply:
:+1: